File: challenges/3-tipos-de-triangulo/ruby/peterjpd/desafio-3.rb
lados = ARGV
lados.map! { |number| number.to_f }.sort!.reverse!
A = lados[0]
B = lados[1]
C = lados[2]

  
if A >= (B + C)
  puts "Não forma triângulo"
else
  if (A**2) == ((B**2) + (C**2))
    puts "Triângulo Retangulo"
  end
  if (A**2) > ((B**2) + (C**2))
    puts "Triângulo Obtusangulo"
  end
  if (A**2) < ((B**2) + (C**2))
    puts "Triângulo Acutangulo"
  end
  if A == B && B == C
    puts "Triângulo Equilátero"
  elsif A == B || B == C || A == C
    puts "Triangulo Isosceles"
  end
end

File: challenges/3-tipos-de-triangulo/javascript/JoaoSortudo/desafio3.js


let a = Number(process.argv.slice(2,3)); 
let b = Number(process.argv.slice(3,4));
let c = Number(process.argv.slice(4,5));

    if (a >= b+c || b >= a + c || c >= a + b){
        console.log("Não forma triângulo.")
    }
    else {
        if (a**2 == b**2 + c**2 || b**2 == a**2 + c**2 || c**2 == a**2 + b**2){
            console.log("Triângulo retângulo.")
        }
        else if (a**2 > b**2 + c**2 || b**2 > a**2 + c**2 || c**2 > a**2 + b**2){
            console.log("Triângulo obtusângulo.")
        }
        else if (a**2 < b**2 + c**2 || b**2 < a**2 + c**2 || c**2 < a**2 + b**2){
            console.log("Triângulo acutângulo.")
        }
      
        if (a === b && a === c){
            console.log("Triângulo equilátero.")
        }
        else if (a === b || a === c || b === c){
            console.log("Triângulo isósceles.")
        }        
    }